File: CourseRegistrationSystem/src/Domain/Staff.java
package Domain;

import java.util.Date;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
@JsonIgnoreProperties(ignoreUnknown = true)
public abstract class Staff extends Person {

    private String staffNo;

    public Staff(String FName, String LName, Date birthdate, String staffNo) {
        super(FName, LName, birthdate);
        setStaffNo(staffNo);
    }

    public Staff(){}

    public String getStaffNo() {
        return staffNo;
    }

    public void setStaffNo(String staffNo) {
        this.staffNo = staffNo;
    }

    @Override
    public String toString() {
        return "Staff{" +
                "staffNo='" + staffNo + '\'' +
                "} " + super.toString();
    }
}

File: Iteration3/Domain/Staff.py
from abc import ABC
from Domain.Person import Person

class Staff(Person, ABC):
    def __init__(self, first_name, last_name, staff_no):
        super().__init__(first_name, last_name)
        self.set_staff_no(staff_no)

    def get_staff_no(self):
        return self.staff_no

    def set_staff_no(self, staff_no):
        self.staff_no = staff_no

    def __str__(self):
        return f"Staff{{staff_no='{self.staff_no}'}} {super().__str__()}"
